frontend/apps/filemanager/filemanagerutil.lua

```diff
@@ -202,14 +202,21 @@ function filemanagerutil.genResetSettingsButton(doc_settings_or_file, caller_cal
                         custom_cover_file    = check_button_cover.checked and custom_cover_file,
                         custom_metadata_file = check_button_metadata.checked and custom_metadata_file,
                     }
-                    (doc_settings or DocSettings:open(file)):purge(nil, data_to_purge)
-                    if data_to_purge.custom_cover_file or data_to_purge.custom_metadata_file then
-                        UIManager:broadcastEvent(Event:new("InvalidateMetadataCache", file))
-                    end
+                    doc_settings = doc_settings or DocSettings:open(file)
                     if data_to_purge.doc_settings then
+                        if doc_settings:has("pagemap_chars_per_synthetic_page") then
+                            local cache_file_path = doc_settings:readSetting("cache_file_path")
+                            if cache_file_path then
+                                os.remove(cache_file_path)
+                            end
```

Comment:
Is this really required ?
What about with the alternative ToC ?
Couldn't/shouldn't you check everything is sane and consistent at opening time ? ie. when the docsettings are sync'ed from another device and there is an existing cre cache file?